### ISSUE TITLE
OpenCV build failed using make -j7. errors in apps > traincascade various cpp files

### ISSUE THREAD

**Issue by iruriksl** [opencv/opencv#18104]
##### System information (version)
- OpenCV => 4.2 master
- Operating System / Platform => macOS Catalina 10.15.5
- Compiler => Cmake 3.18.1; GNU Make 3.81
##### Detailed description

trying to build OpenCV but getting many errors when run "make -j7". The errors are in train cascade apps folder files.

##### Steps to reproduce
Downlaod master code. remove comments for create cascade and traincascade in Cmake file of apps directory. Configure and Generate using CMake Gui. Run command make -j7 and the error starts generating.
Tried rectifying them in "old_ml.hpp" for CvFileStorage by using cv::FileStorage and so on. But on correcting many different errors occur in many of the files in this folder.

Few of the errors extract:
<!--
Scanning dependencies of target example_cpp_videocapture_image_sequence
[ 89%] Building CXX object samples/cpp/CMakeFiles/example_cpp_videocapture_image_sequence.dir/videocapture_image_sequence.cpp.o
In file included from /Users/pinkss/opt/opencv/opencv/apps/traincascade/old_ml_precomp.hpp:45:\

\f1\b /Users/pinkss/opt/opencv/opencv/apps/traincascade/old_ml.hpp:803:24: \cf5 error: \cf2 unknown type name 'CvFileStorage'; did you mean
\f0\b0 \

\f1\b       'cv::FileStorage'?
\f0\b0 \
    virtual void read( CvFileStorage* fs, CvFileNode* node );\

\f1\b \cf4                        ^~~~~~~~~~~~~
\f0\b0 \cf2 \
\cf4                        cv::FileStorage\cf2 \

\f1\b /Users/pinkss/opt/opencv/opencv/modules/core/include/opencv2/core/persistence.hpp:303:20: note: 
\f0\b0 'cv::FileStorage' declared\
      here\
class CV_EXPORTS_W FileStorage\

\f1\b \cf4                    ^
\f0\b0 \cf2 \
In file included from /Users/pinkss/opt/opencv/opencv/apps/traincascade/old_ml_boost.cpp:41:\
In file included from /Users/pinkss/opt/opencv/opencv/apps/traincascade/old_ml_precomp.hpp:45:\

/Users/pinkss/opt/opencv/opencv/apps/traincascade/cascadeclassifier.cpp:373:28: error: no member named 'fs' in
      'cv::FileStorage'
        cvWriteComment( fs.fs, cmnt, 0 );
                        ~~ ^
/Users/pinkss/opt/opencv/opencv/apps/traincascade/boost.cpp:1020:20: error: cannot decrement value of type
      'cv::FileNodeIterator'
    internalNodesIt--; leafValsuesIt--;
    ~~~~~~~~~~~~~~~^
In file included from /Users/pinkss/opt/opencv/opencv/apps/traincascade/old_ml_boost.cpp:41:
-->


[build_error.txt](https://github.com/opencv/opencv/files/5080128/build_error.txt)

**Comment by alalek** [comment]
> remove comments for create cascade and traincascade

They are commented for reason - not supported anymore.
Use OpenCV 3.4 instead.

duplicate of #13231

---

Build / usage questions should go to Users OpenCV Q/A forum: http://answers.opencv.org

**Comment by dogbaha** [comment]
I got the same problem , and I fix it by using opecv-3.4.5. Write the method if you found a better way to get createcascade and createsamples under opencv-4+
----
os: centos 8
gcc : 8.3 c++11
cmake : 3.11
